docs/reference/ingest/processors/registered-domain.asciidoc

```diff
@@ -0,0 +1,83 @@
+[role="xpack"]
+[testenv="basic"]
+[[registered-domain-processor]]
+=== Registered domain processor
+++++
+<titleabbrev>Registered Domain</titleabbrev>
+++++
+
+Extracts the registered domain (also known as effective top-level domain or
+eTLD), sub-domain, and top-level domain from a fully qualified domain name
+(FQDN). Uses the registered domains defined in the
+https://publicsuffix.org/[Mozilla Public Suffix List].
+
+[[registered-domain-options]]
+.Registered Domain Options
+[options="header"]
+|======
+| Name               | Required | Default | Description
+| `field`            | yes      |         | Field containing the source FQDN.
+| `target_field`     | no       | `<empty string>` | Field under which the
+extracted domain components will be written. If the empty string, the domain
+components will be added to the root of the document.
```

Comment:
Just some minor rewording here. Feel free to ignore if wanted. It seemed odd to me that `url` wasn't the default to fit ECS, but there's probably a good reason for that.
```suggestion
| `target_field`     | no       | `<empty string>` | Object field containing
extracted domain components. If an `<empty string>`, the processor adds
components to the document's root.
```

Reply:
> Just some minor rewording here. Feel free to ignore if wanted. It seemed odd to me that `url` wasn't the default to fit ECS, but there's probably a good reason for that.

Fyi, the reason for that is that there are multiple places in ECS where registered domain info may be populated so we didn't want to default to just one of them. See https://github.com/elastic/elasticsearch/pull/67611#discussion_r603368001 for more context.

Reply:
Makes sense. Thanks for the explanation @danhermann!